docs/guides/LiveCode Builder Bytecode Reference.md

```diff
@@ -0,0 +1,154 @@
+---
+group: reference
+---
+
+# LiveCode Builder Bytecode Reference
+
+## Introduction
+
+The LiveCode Builder Virtual Machine is a infinite register machine with a
+high-level and verifiable bytecode.
+
+All bytecode runs in the context of a module instance with executing handlers
+forming a stack of activation frames. Each frame contains an array of registers,
+the first part of which are parameters followed by handler-local variables and
+bytecode block registers.
+
+Most bytecode operations operate directly on registers, access to module level
+definitions (handlers, constants, variables) are indirected through the fetch
+and store operations.
+
+Each bytecode operation has an address which can be jumped to using the jump
+operations.
+
+## Operations
+
+### Jump
+
+    jump <label>
+
+The jump operation sets the address of the next instruction to execute to
+that identified by <label>.
+
+### Jump If False
+
+    jump_if_false <register>, <label>
+
+The jump_if_true operation checks the value in <register> and jumps to <label>
+if it is 'false'.
+
+If is a runtime error if <register> does not contain a boolean.
+
+### Jump If True
+
+    jump_if_true <register>, <label>
+
+The jump_if_true operation checks the value in <register> and jumps to <label>
+if it is 'true'.
+
+If is a runtime error if <register> does not contain a boolean.
+
+### Assign Constant
+
+    assign_constant <register>, <constant>
+
+The assign_constant operation copies <constant> into <register>. Here <constant>
+can be 'nothing', a boolean, integer, real, string, list or array.
+
+If <register> is typed then it is a runtime error if the type of <constant>
+does not conform to that type.
+
+### Assign
+
+    assign <dst-register>, <src-register>
+
+The assign operation copies the value in <src-register> to <dst-register>.
+
+If <dst-register> is typed then it is runtime error if the type of the value
+in <src-register> does not conform to that type.
+
+### Return
+
+    return [ <result-reg> ]
+
+The return operation exits the current handler, returning to the instruction after
+the invoke operation which created it. If present, the value in <result-reg> is
+copied as the return value. If not present then the return value will be 'nothing'.
+Additionally any values in out or inout parameters are copied back to the caller.
+
+If <result-reg> is not present, then the return value will be 'nothing'.
+
+It is a runtime error if the type of the return value does not conform to the
+return type of the current handler's signature.
+
+It is a runtime error if any out or inout parameters are unassigned at the point
+of return.
+
+### Invoke
+
+    invoke <handler>, <result>, <arg_1>, ..., <arg_n>
+
+The invoke operation creates a new frame copying values from the argument registers
+for any in or inout parameters. It then starts executing the bytecode attached
+to <handler>.
```

Comment:
It might be worth mentioning that the `<handler>` needs to be a definition and that `<result>` is the register into which the return value from the invocation is placed (and the types need to match).